swift/Picture.swift

```diff
@@ -113,13 +113,11 @@ class Picture {
         let size = getSize()
         let xRatio = size.width / rect.width
         let yRatio = size.height / rect.height
-
+        
         // Scale the size of the picture relative to that ratio.
-        let width = size.width * xRatio
-        let height = size.height * yRatio
-        resize(CGSize(width: width, height: height))
+        apply(transform: CGAffineTransform(scaleX: xRatio, y: yRatio), anchorPoint: CGPoint(x: 0, y: 0))
```

Comment:
The way you're using transforms is getting out of hands. You call `apply(transform:)` from different unrelated places, and for your logic to work you need to make sure those calls are in the correct order. Here's the thing though, that correct order is not explicitly specified anywhere. You just need to put the calls in the place where they work. 

Let me explain better. This is your code inside the `render` function:

```swift
public func render(...,  contentRect: CGRect, rotation: Double) throws {
   ...
  picture.resetTransform()
  picture.stretchToFit(contentRect)
  picture.apply(transform: CGAffineTransform(rotationAngle: rotation * .pi / 180.0))
  ...
}
```

If you change the order of those calls, your code will break. And it's not necessarily obvious why they need to happen in that order. 

If all you're doing inside `stretchToFit` is setting a transform, you might as well return that transform instead of setting it yourself directly. Your code would become something like this:

```swift
public func render(...,  contentRect: CGRect, rotation: Double) throws {
   ...
  let transform = picture.transformForStretchingToFill(contentRect) // probably needs a better name
      .concatenating(CGAffineTransform(rotationAngle: rotation * .pi / 180.0))
  picture.apply(transform: transform)
  ...
}
```

Now the correct order of the transforms is explicit, and you can also get rid of the `resetTransform` function. Now `apply(transform:)` would override the previous transform. This change would help us get closer to stop relying on previously stored state.